### ISSUE TITLE
Field validation at data import (not at mapping)

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#85]
Right now (issue # 81, with commit # 84) for records with fields that have too many characters, we are truncating the data (at the time that the building snap shot is created. It would be better to somehow identify those problem records for the user so they can fix the data and re-import if necessary.

This will need some design discussion before implementing.

**Comment by RDmitchell** [comment]
See the comments in Issue # 84, with many suggestions about making this error handling much more robust.

**Comment by RDmitchell** [comment]
Seems like one place to start is with validating the fields (the non JSON fields that is) against their definitions in the BuildingSnapShot database table. 

For example, in a sample import file, I set the field for site_eui in the db, which is defined as a float, to a very long string. After allowing me to map that field, the program completely dropped the string, so now the field is blank in the canonical building record, with no indication from the program what the problem was with the data. 

After importing the file, the mapping screen shows the very long string in the Site EU field, which is defined in the db as a float
![image](https://cloud.githubusercontent.com/assets/6314950/7186379/2279f756-e41e-11e4-9b60-047140cd5878.png)

During the mapping process, the program drops the data in the field for Site EUI, never tells the user what the problem was. 

![image](https://cloud.githubusercontent.com/assets/6314950/7186429/847d8daa-e41e-11e4-8842-417e2bf3fbf9.png)

**Comment by RDmitchell** [comment]
The file used in the example above
https://drive.google.com/a/lbl.gov/folderview?id=0B3fTKpZ9Dx7LNVBoVExWUWdscHc&usp=sharing

**Comment by rebrownlbnl** [comment]
This is a subset of the functionality needed for data cleansing, issue #177.

**Comment by RDmitchell** [comment]
@nllong -- don't know if this is something we want to put on our plates for the Sept 2016 release or just move it to FY 2017

**Comment by RDmitchell** [comment]
@macintoshpie -- I'm wondering if this should be reopened, put into Test, assigned to me, and I can test to see if this is still and issue, and then close if it's no longer a problem, or keep it open if it is... does that make sense?

**Comment by macintoshpie** [comment]
@RDmitchell sounds good

**Comment by macintoshpie** [comment]
Gotcha. As a quick note, my hunch on what we're seeing above is a data-type issue, where we try to parse the string as a number and when we fail we set it to NULL.
Then we run the validation (post-type-coersion) and find that it's NULL (not that it's a string).

SEED's DQ check _does_ catch that the value is null. But it might not be obvious to the user _why_ it's saying it's null (because they put an invalid data type, e.g. string, into the column which expects a number).

I'm not sure how we could catch this issue earlier (the original request of this ticket). I would think we'd need to know what data type we expect a spreadsheet column to be before we can validate it. And users indicate what the data type is on the mapping page.

@RDmitchell am I understanding this correctly? Can you think of how we could validate data before users "map" the data (telling us what the data-types are)?

**Comment by RDmitchell** [comment]
@macintoshpie -- right, I'm not sure what the solution is. 

But right now the program isn't really validating whether data in a field matches the field type, which might be something to add somewhere. 

This is probably a low priority issue but seems like at some point we should do something about it? 

Or you all can decide to not deal with it and just close it, as it doesn't seem like any users are running into this problem
